sdk/python/tests/integration/feature_repos/universal/data_sources/file.py

```diff
@@ -352,3 +360,73 @@ def create_offline_store_config(self):
             staging_location_endpoint_override=self.endpoint_url,
         )
         return self.duckdb_offline_store_config
+
+
+class RemoteOfflineStoreDataSourceCreator(FileDataSourceCreator):
+    def __init__(self, project_name: str, *args, **kwargs):
+        super().__init__(project_name)
+        self.server_port: int = 8815
+        self.proc: Process = None
+
+    def setup(self, registry: RegistryConfig):
+        parent_offline_config = super().create_offline_store_config()
+
+        self.server_port = free_port()
+        host = "0.0.0.0"
+        self.proc = Process(
+            target=start_test_local_server,
+            args=(
+                parent_offline_config,
+                self.project_name,
+                registry.path,
+                host,
+                self.server_port,
+            ),
+        )
+        self.proc.start()
+        _time_out_sec: int = 180
+
+        # Wait for server to start
+        wait_retry_backoff(
+            lambda: (None, check_port_open(host, self.server_port)),
+            timeout_secs=_time_out_sec,
+            timeout_msg=f"Unable to start the feast server in {_time_out_sec} seconds for remote offline store type, port={self.server_port}",
+        )
+        logger.info("Port has been freed.")
+        return "grpc+tcp://{}:{}".format(host, self.server_port)
+
+    def create_offline_store_config(self) -> FeastConfigBaseModel:
+        self.remote_offline_store_config = RemoteOfflineStoreConfig(
+            type="remote", host="0.0.0.0", port=self.server_port
+        )
+        return self.remote_offline_store_config
+
+    def teardown(self):
+        if self.proc is None and self.proc.is_alive():
+            self.proc.kill()
+            logger.info("Server process killed.")
+
+            # wait server to free the port
+            wait_retry_backoff(
+                lambda: (
+                    None,
+                    not check_port_open("localhost", self.server_port),
+                ),
+                timeout_secs=10,
+            )
+        super().teardown()
+
+
+def start_test_local_server(
+    parent_offline_config, project_name, registry_path, host, port
+):
+    fs = FeatureStore(
+        config=RepoConfig(
+            project=project_name,
+            provider="local",
+            offline_store=parent_offline_config,
+            registry=registry_path,
+            entity_key_serialization_version=2,
+        )
+    )
+    start_server(fs, host, port)
```

Comment:
I might be grasping for straws, but maybe we can try `fs.serve_offline` instead of `start_server` to avoid an import. If that doesn't work, how about we simply prepopulate some temp folder with feature_store.yaml and then start offline server with cli command pointing to that feature_store.yaml.

Reply:
you can use this `yaml.dump(repo_config.dict(by_alias=True))` to generate file contents

Reply:
@tokoko Thanks you mean something like this 

```
    feature_store_configuration = yaml.dump(self.repo_config.dict(by_alias=True))
    fs.repo_path= feature_store_configuration
    fs.serve_offline(host, port)
```

Reply:
No, in that case I meant something like this, use `subprocess` instead of `multiprocessing`:
```
with open('temp.yml', 'w') as outfile:
    yaml.dump(self.repo_config.dict(by_alias=True), outfile)
    
process = subprocess.Popen(['feast', 'serve_offline', ...]) 
```

Reply:
@tokoko Thanks. I have updated the [code](https://github.com/feast-dev/feast/pull/4262/commits/08c2616e6e4972ed0ffe3e85e1379ff0e96ac786) and test are passing.